src/GraphManager/GraphEdit/ButtonTests.test.tsx

```diff
@@ -0,0 +1,178 @@
+      ctrl.mode.setAllowGraphInteractions((current: boolean) => !current);
+  
+      // Assert: Check that graph interactions are now enabled
+      expect(ctrl.mode.allowGraphInteractions).toBe(true);
+      expect(ctrl.mode.setAllowGraphInteractions).toHaveBeenCalledTimes(1);
+  
+      // Act: Simulate button click to disable graph interactions
+      ctrl.mode.setAllowGraphInteractions((current: boolean) => !current);
+  
+      // Assert: Check that graph interactions are now disabled
+      expect(ctrl.mode.allowGraphInteractions).toBe(false);
+      expect(ctrl.mode.setAllowGraphInteractions).toHaveBeenCalledTimes(2);
+    });
+  
+  });
+
+  describe("EditModeButton", () => {
+    it("should toggle ctrl.mode.isEditingEnabled when clicked", () => {
+      // Arrange: Mock controller
+      const ctrl = makeMockController();
+      ctrl.mode.isEditingEnabled = false; // Initial state
+      ctrl.mode.setIsEditingEnabled = jest.fn((newValue: boolean) => {
+        ctrl.mode.isEditingEnabled = newValue;
+      });
+  
+      // Act: Simulate enabling edit mode
+      ctrl.mode.setIsEditingEnabled(!ctrl.mode.isEditingEnabled);
+  
+      // Assert: Check that edit mode is now enabled
+      expect(ctrl.mode.isEditingEnabled).toBe(true);
+      expect(ctrl.mode.setIsEditingEnabled).toHaveBeenCalledWith(true);
+      expect(ctrl.mode.setIsEditingEnabled).toHaveBeenCalledTimes(1);
+  
+      // Act: Simulate disabling edit mode
+      ctrl.mode.setIsEditingEnabled(!ctrl.mode.isEditingEnabled);
+  
+      // Assert: Check that edit mode is now disabled
+      expect(ctrl.mode.isEditingEnabled).toBe(false);
+      expect(ctrl.mode.setIsEditingEnabled).toHaveBeenCalledWith(false);
+      expect(ctrl.mode.setIsEditingEnabled).toHaveBeenCalledTimes(2);
+    });
+  
+    // it("should show an alert if user is not logged in when clicked", () => {
+    //     // Code here
+    //       });
+    
+    //   it("should render the correct icon based on isEditingEnabled state", () => {
+    //    //code here
+    //   });   
+  });
+
+describe("CreateButton", () => {
+    let mockController: any;
+
+    beforeEach(() => {
+        mockController = {
+          mode: { isEditingEnabled: true },
+        };
+      });
+
+  it("should update anchorEl state when button is clicked", () => {
+    // Arrange: Mock the initial state of anchorEl
+    let anchorEl: HTMLElement | null = null;
+
+    // Simulate the setAnchorEl function that changes the anchorEl state
+    const setAnchorEl = (value: HTMLElement | null) => {
+      anchorEl = value;
+    };
+
+    // Act: Simulate button click by manually updating the anchorEl state
+    setAnchorEl(document.createElement("div")); // Simulating the button click setting anchorEl
+
+    // Assert: Check if anchorEl is set correctly
+    expect(anchorEl).not.toBeNull(); // Assert that anchorEl is set to a non-null value
+  });
+
+  it("should call openCreateNodePopUpAtPagePosition when 'New Node' is selected", () => {
```

Comment:
also nothing tested here: our function is called yes, but the result is not verified, the only expect() statement here is on the input of the function, which is passed in this test, so nothing is verified in our code